TheSadRogue.Integration/Maps/RogueLikeMapBase.IScreenObject.cs

```diff
@@ -35,20 +35,47 @@ public abstract partial class RogueLikeMapBase
         bool IScreenObject.HasSadComponent<TComponent>(out TComponent component) => BackingObject.HasSadComponent(out component);
 
         /// <inheritdoc/>
-        public bool ProcessKeyboard(Keyboard keyboard) => BackingObject.ProcessKeyboard(keyboard);
+        public virtual bool ProcessKeyboard(Keyboard keyboard) => BackingObject.ProcessKeyboard(keyboard);
 
         /// <inheritdoc/>
-        public bool ProcessMouse(MouseScreenObjectState state) => BackingObject.ProcessMouse(state);
+        bool IScreenObject.ProcessMouse(MouseScreenObjectState state)
+        {
+            // Transformation is required to ensure that backing objects which are surfaces are handled correctly.
+            // See https://github.com/thesadrogue/TheSadRogue.Integration/issues/47.
+            state = new MouseScreenObjectState(BackingObject, state.Mouse.Clone());
+            return ProcessMouse(state);
+        }
+
+        /// <summary>
+        /// Overridable implementation of IScreenObject's ProcessMouse which is guaranteed to receive a mouse state
+        /// appropriate for the map implementation.
+        /// </summary>
+        /// <param name="state"/>
+        /// <returns/>
+        protected virtual bool ProcessMouse(MouseScreenObjectState state) => BackingObject.ProcessMouse(state);
 
         /// <inheritdoc/>
-        public void LostMouse(MouseScreenObjectState state) => BackingObject.LostMouse(state);
+        void IScreenObject.LostMouse(MouseScreenObjectState state)
+        {
+            // Transformation is required to ensure that backing objects which are surfaces are handled correctly.
+            // See https://github.com/thesadrogue/TheSadRogue.Integration/issues/47.
+            state = new MouseScreenObjectState(BackingObject, state.Mouse.Clone());
+            LostMouse(state);
+        }
+
+        /// <summary>
+        /// Overridable implementation of IScreenObject's LostMouse which is guaranteed to receive a mouse state
+        /// appropriate for the map implementation.
+        /// </summary>
+        /// <param name="state"/>
+        protected virtual void LostMouse(MouseScreenObjectState state) => BackingObject.LostMouse(state);
```

Comment:
This one I think you don't need to actually transform it. I believe (scanning through the code yesterday but not really playing with it) I noticed that the state sent to `LostMouse` is the state for whatever object stole the mouse, or in the case of leaving the window, I believe it's a state that has a `null` screen object reference. So this can be restored and just made virtual

```suggestion
        public virtual void LostMouse(MouseScreenObjectState state) => BackingObject.LostMouse(state);
```